bugsnag-android-core/src/test/java/com/bugsnag/android/AppDataCollectorSerializationTest.kt

```diff
@@ -74,8 +74,17 @@ internal class AppDataCollectorSerializationTest {
     @Parameter
     lateinit var testCase: Pair<App, String>
 
+    private val eventMapper = BugsnagEventMapper(NoopLogger)
+
     @Test
     fun testJsonSerialisation() {
         verifyJsonMatches(testCase.first, testCase.second)
     }
+
+    @Test
+    fun testJsonDeserialization() {
+        verifyJsonParser(testCase.first, testCase.second) {
+            eventMapper.convertAppWithState(it)
+        }
```

Comment:
What does this test actually do? I'm sure I'm missing context here from various helpers, and its nice that its concise, but it seems unclear what the `first` and `second` and `it` are to determine if this covers everything in the changeset.

Reply:
The helper works by

1. parsing the json fixture with the `convertAppWithState` function
2. turning the in-memory model back into json using the existing `toStream(writer: JsonStream)` function
3. comparing the output of the in-memory model with the test fixture (they should be equal)

It could definitely be a bit clearer. I'll add some named parameters to help.

Reply:
Ah, thanks for that, makes sense!